src/utils/material.js

```diff
@@ -10,24 +11,42 @@ module.exports.updateMapMaterialFromData = function (materialName, dataName, sha
   var el = shader.el;
   var material = shader.material;
   var src = data[dataName];
+  // Because a single material / shader may have multiple textures,
+  // we need to remember the source value for this data property
+  // to avoid redundant operations which can be expensive otherwise
+  // (e.g. video texture loads).
   var shadowSrcName = '_texture_' + dataName;
 
-  if (src) {
-    if (src === shader[shadowSrcName]) { return; }
-    // Texture added or changed.
-    shader[shadowSrcName] = src;
-    if (src instanceof THREE.Texture) { setMap(src); return; }
-    el.sceneEl.systems.material.loadTexture(src, {src: src, repeat: data.repeat, offset: data.offset, npot: data.npot}, setMap);
+  if (!src) {
+    // Forget the prior material src.
+    shader[shadowSrcName] = null;
```

Comment:
Would not be better to store all the material src within an object and not hanging directly from the shader with a strange `__texture__` key? Something like `shader.materialsSrc = []`